lib/vicadmin/validate.go

```diff
@@ -172,20 +174,42 @@ func NewValidator(ctx context.Context, vch *config.VirtualContainerHostConfigSpe
 
 	// Network Connection Check
 	hosts := []string{
-		"google.com:80",
-		"docker.io:443",
+		"http://google.com",
+		"https://docker.io",
 	}
+
 	nwErrors := []error{}
 
-	for _, host := range hosts {
-		conn, err := net.DialTimeout("tcp", host, time.Minute/2)
+	// create a http client with a custom transport. The transport uses the [s/h]proxy defined in vicmachine create if applicable.
+	client := &http.Client{Timeout: 30 * time.Second}
+	vchArgs := make(map[string]string)
+	for _, args := range vch.VicMachineCreateOptions {
+		splitArgs := strings.Split(args, "=")
+		vchArgs[splitArgs[0]] = splitArgs[1]
+	}
+	//priority given to https proxies
+	proxy := ""
+	if sproxy := vchArgs["--https-proxy"]; sproxy != "" {
```

Comment:
Additionally, we will likely want to check if the element is actually present in the map. It is true that even when an element is not present in a `map[string]string`the return value witll be `""`. However, it is still of interest of us whether or not the value was set to that or just defaulted to that value. 

```
 proxy, ok := vchArgs["--https-proxy"]
if !ok {
 	proxy = vchArgs["--http-proxy"]
}
```
when retrieving values from a map if `golang` you always get two values back. the value and a boolean for if the key was present.